components/common.js

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable */
+
+import { PropTypes } from 'react';
+import {
+  requireNativeComponent,
+  NativeModules,
+  Platform,
+} from 'react-native';
+
+export const SUPPORTED = 'SUPPORTED';
+export const USES_DEFAULT_IMPLEMENTATION = 'USES_DEFAULT_IMPLEMENTATION';
+export const NOT_SUPPORTED = 'NOT_SUPPORTED';
+
+export function airMapName(mapProvider) {
+  if (mapProvider === 'google') return 'AIRGoogleMap';
+  return 'AIRMap';
+}
+
+export function airComponentName(mapProvider, component) {
+  return `${airMapName(mapProvider)}${component}`;
+}
+
+export const contextTypes = {
+  mapProvider: PropTypes.string,
+};
+
+export const createNotSupportedComponent = message => () => console.error(message) || null;
+
+export const AIRGoogleMapIsInstalled = !!NativeModules.UIManager[airMapName('google')];
+
+export function decorateMapComponent(Component, { componentType, providers }) {
+  const components = {
+    default: requireNativeComponent(airComponentName(null, componentType), Component),
```

Comment:
What do you think about creating this on-demand, only if necessary?  It seems a little wasteful to create it even if it's not used.